crates/stages/src/stages/sender_recovery.rs

```diff
@@ -162,11 +172,25 @@ impl<DB: Database> Stage<DB> for SenderRecoveryStage {
         let latest_tx_id = tx.block_body_indices(unwind_to)?.last_tx_num();
         tx.unwind_table_by_num::<tables::TxSenders>(latest_tx_id)?;
 
+        let stage_checkpoint = stage_progress(tx)?;
+
         info!(target: "sync::stages::sender_recovery", to_block = input.unwind_to, unwind_progress = unwind_to, is_final_range, "Unwind iteration finished");
-        Ok(UnwindOutput { checkpoint: StageCheckpoint::new(unwind_to) })
+        Ok(UnwindOutput {
+            checkpoint: StageCheckpoint::new(unwind_to)
+                .with_entities_stage_checkpoint(stage_checkpoint),
+        })
     }
 }
 
+fn stage_progress<DB: Database>(
```

Comment:
should we define this on the trait level once we implement it for all of the stages?

Reply:
what definition on trait level do you have in mind? maybe I misunderstood

Reply:
just in general, every stage must implement `stage_progress`/`stage_checkpoint` method. the implementation details are up to the respective stage